Cabal/Cabal.cabal

```diff
@@ -342,6 +342,14 @@ library
   if !impl(ghc >= 7.0)
     default-extensions: CPP
 
+executable cabal-setup
```

Comment:
Yep, that's what I had I mind. Is Cabal smart enough to see that `--enable-tests` must imply `cabal-setup-buildable: True`? Then maybe the second conditional could be elided.